pdal/plang/Environment.cpp

```diff
@@ -279,31 +279,38 @@ std::string readPythonString(PyObject* list, Py_ssize_t index)
 
     return ss.str();
 }
-void addMetadata(PyObject *list, MetadataNode m)
+void addMetadata(PyObject *dict, MetadataNode m)
 {
 
-    if (!PyList_Check(list))
+    if (! dict)
+    {
         return;
+    }
 
-    for (Py_ssize_t i = 0; i < PyList_Size(list); ++i)
-    {
-        PyObject *tuple = PyList_GetItem(list, i);
-        if (!PyTuple_Check(tuple) || PyTuple_Size(tuple) != 5)
-            continue;
+    if (!PyDict_Check(dict) )
+        throw pdal::pdal_error("'metadata' member must be a dictionary!");
```

Comment:
I've been trying to make sure that all errors have a stage name attached if possible:

getName() + ": blah"

Perhaps we should standardize.